File: javascript/485_Max_Consecutive_Ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
  let maxConsecutive = 0;
  let tempMax = 0;
  let i = 0;
  let j = 0;
  
  for (let num of nums) {
    if (num === 1) {
      tempMax += 1;
    } else {
      maxConsecutive = tempMax > maxConsecutive ? tempMax : maxConsecutive;
      tempMax = 0;
    }
  }

  maxConsecutive = tempMax > maxConsecutive ? tempMax : maxConsecutive;  
  
  return maxConsecutive;
};

File: python/485_Max_Consecutive_ones.py
class Solution:
  def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
    maxConsecutive: int = 0
    currMax: int = 0
    
    for num in nums:
      if num == 1:
        currMax += 1
      else:
        maxConsecutive = max(maxConsecutive, currMax)
        currMax = 0
    
    return max(maxConsecutive, currMax)
    